test/evergreen.yml

```diff
@@ -2640,7 +2587,59 @@ buildvariants:
       LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libeatmydata.so PATH=/opt/mongodbtoolchain/v3/bin:$PATH LD_LIBRARY_PATH=$(pwd)/.libs top_srcdir=$(pwd)/.. top_builddir=$(pwd)
   tasks:
     - name: ".pull_request !.windows_only !.pull_request_compilers !.python"
-    - examples-c-test
+    - name: make-check-test
+    - name: examples-c-test
+
+- name: ubuntu1804-msan
+  display_name: "! Ubuntu 18.04 MSAN"
+  run_on:
+  - ubuntu1804-test
+  expansions:
+    configure_env_vars:
+      # We don't compile C++ for memory sanitized testing as it creates false positives.
+      CC=/opt/mongodbtoolchain/v3/bin/clang
+      PATH=/opt/mongodbtoolchain/v3/bin:$PATH
+      CFLAGS="-fsanitize=memory -fno-omit-frame-pointer -ggdb"
+    posix_configure_flags:
+      --enable-silent-rules --enable-strict --enable-diagnostic
+      --disable-static --prefix=$(pwd)/LOCAL_INSTALL
+    smp_command: -j $(grep -c ^processor /proc/cpuinfo)
+    make_command: PATH=/opt/mongodbtoolchain/v3/bin:$PATH make
+    test_env_vars:
+      MSAN_OPTIONS=abort_on_error=1:disable_coredump=0
+      MSAN_SYMBOLIZER_PATH=/opt/mongodbtoolchain/v3/bin/llvm-symbolizer
+      TESTUTIL_SLOW_MACHINE=1
+      LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libeatmydata.so LD_LIBRARY_PATH=$(pwd)/.libs
+      PATH=/opt/mongodbtoolchain/v3/bin:$PATH top_srcdir=$(pwd)/.. top_builddir=$(pwd)
+  tasks:
+    - name: ".pull_request !.windows_only !.pull_request_compilers !.python !.cppsuite"
+    - name: make-check-msan-test
+    - name: examples-c-test
+
+- name: ubuntu1804-ubsan
+  display_name: "! Ubuntu 18.04 UBSAN"
+  run_on:
+  - ubuntu1804-test
+  expansions:
+    configure_env_vars:
+      CC=/opt/mongodbtoolchain/v3/bin/clang
+      CXX=/opt/mongodbtoolchain/v3/bin/clang++
+      PATH=/opt/mongodbtoolchain/v3/bin:$PATH
+      CFLAGS="-fsanitize=undefined -fno-omit-frame-pointer -ggdb"
+      CXXFLAGS="-fsanitize=undefined -fno-omit-frame-pointer -ggdb"
+    posix_configure_flags:
+      --enable-silent-rules --enable-strict --enable-diagnostic
+      --disable-static --prefix=$(pwd)/LOCAL_INSTALL
+    smp_command: -j $(grep -c ^processor /proc/cpuinfo)
+    make_command: PATH=/opt/mongodbtoolchain/v3/bin:$PATH make
+    test_env_vars:
+      UBSAN_OPTIONS="print_stacktrace=1:external_symbolizer_path=/opt/mongodbtoolchain/v3/bin/llvm-symbolizer:detect_leaks=1:abort_on_error=1:disable_coredump=0"
```

Comment:
`halt_on_error=1` option was set in the removed `ubsan-test` task, should we include it here?

Reply:
Fixed.